src/main/java/ecdar/controllers/CanvasController.java

```diff
@@ -145,11 +145,17 @@ private void onActiveModelChanged(final HighLevelModelPresentation oldObject, fi
             modelPane.getChildren().add(activeComponentPresentation);
 
             // To avoid NullPointerException on initial model
-            if (oldObject != null) zoomHelper.resetZoom();
+            Platform.runLater(zoomHelper::resetZoom);
 
         } else if (newObject instanceof DeclarationsPresentation) {
             activeComponentPresentation = null;
             modelPane.getChildren().add(newObject);
+
+            // Bind size of Declaration to size of the model pane to ensure alignment and avoid drag
+            ((DeclarationsController) newObject.getController()).root.minWidthProperty().bind(modelPane.minWidthProperty());
```

Comment:
We should consider moving ```((DeclarationsController) newObject.getController())``` into a variable declaration such as 
```
DeclarationsController controller = (DeclarationsController)newObject.getController();
controller.root.minWidthProperty.bind(modelPane.minWidthProperty()));
...
```
Or
```
Controller controller = newObject.getController();
if (!(controller instaneof DeclarationsController)) {
  throw ...
}
var root = ((DeclarationsController) controller).root;
root.minWidthProperty().bind(modelPane.minWidthProperty());
...
```
I think this would reduce complexity.

Reply:
I agree, this would be nice. I vote for the first option, as we are already sure that newObject is a `DeclarationsPresentation` and must therefore have a `DeclarationsController` (we are vulnerable if the logic does not hold, but this would violate fundamental design decisions)

Reply:
Very good :smiley: